### PR TITLE
rules: declare UserConfigurationSchema as a schema

### PR DESCRIPTION
**Pull request by jd**

### CODE REVIEW

**Comment by jd** [comment]
Pull-request updated, HEAD is now 6d856338c169e6c5df2568b18e54c94f86dc6671

**Comment by jd** [comment]
Pull-request updated, HEAD is now 59237b2000cf7dd05c18367ead52f80e4a98566a

File: mergify_engine/rules.py [hunk 1]
```diff
@@ -50,12 +50,12 @@
...
     }),
     'enforce_admins': voluptuous.Any(None, bool),
-}
+})
 
 # TODO(sileht): We can add some otherthing like
 # automatic backport tag
 # option to disable mergify on a particular PR
-Rule = {
+Rule = voluptuous.Schema({
```
Comment (sileht) [inline mergify_engine/rules.py]: Here [1] `Rule` is already a `Schema` with `required=True`. And voluptuous does not propagate Schema argument:

```python
>>> from voluptuous import Schema
>>> v.Schema({"foo": str}, required=True)({})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/voluptuous/schema_builder.py", line 192, in __call__
    return self._compiled([], data)
  File "/usr/lib/python3/dist-packages/voluptuous/schema_builder.py", line 486, in validate_dict
    return base_validate(path, iteritems(data), out)
  File "/usr/lib/python3/dist-packages/voluptuous/schema_builder.py", line 324, in validate_mapping
    raise er.MultipleInvalid(errors)
voluptuous.error.MultipleInvalid: required key not provided @ data['foo']
>>> v.Schema(v.Schema({"foo": str}), required=True)({})
{}
>>>
```
I was Expecting a failure but with a `Schema` inside a `Schema`, `required=True` no longuer works.

[1] https://github.com/Mergifyio/mergify-engine/pull/160/files#diff-ff2f0c79b43d73c315cb37944cd0c551R98

**Comment by jd** [comment]
Pull-request updated, HEAD is now 6de03d00a4e3d41fac80baf6572c3fb42b9a26be

File: mergify_engine/rules.py [hunk 2]
```diff
@@ -50,13 +50,13 @@
...
     'enforce_admins': voluptuous.Any(None, bool),
-}
+})
 
 # TODO(sileht): We can add some otherthing like
 # automatic backport tag
 # option to disable mergify on a particular PR
 Rule = {
-    'protection': Protection,
+    'protection': ProtectionSchema,
```
Comment (sileht) [inline mergify_engine/rules.py]: What I said for Rule is also valid for anything inside the schema. This make Protection attributes no more required.

**Pull request closed without merging**